ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoSignedTransactionVerifier.kt

```diff
@@ -0,0 +1,34 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.v5.ledger.utxo.UtxoLedgerService
+import java.io.Serializable
+
+/**
+ * [UtxoSignedTransactionVerifier] verifies a [UtxoSignedTransaction].
+ *
+ * Implement [UtxoSignedTransactionVerifier] and pass the implementation into [UtxoLedgerService.receiveFinality] to perform
+ * custom verification on the [UtxoSignedTransaction] received from the initiator of finality.
+ *
+ * When verifying a [UtxoSignedTransaction] throw either an [IllegalArgumentException], [IllegalStateException] or
+ * [CordaRuntimeException] to indicate that the transaction is invalid. This will lead to the termination of finality for the caller of
+ * [UtxoLedgerService.receiveFinality] and all participants included in finalizing the transaction. Other exceptions will still stop
+ * the progression of finality; however, the reason for the failure will not be communicated to the initiator of finality.
+ *
+ * @see UtxoLedgerService.receiveFinality
+ */
+fun interface UtxoSignedTransactionVerifier : Serializable {
```

Comment:
Do we need this in the public API? I've got something similar in the runtime, but I doubt users will ever want to verify manually.

Reply:
@MrMatthewLayton  @lankydan 
For Consensual, we had `ConsensualSignedTransactionVerifier`.
I think this is the extra check what the users can feed in to the `receiveFinality`.
It was called `checkTransaction` in C4.
Is not this needed for us here?

Reply:
I think this is very confusingly named. Is this a callback for the flow to check transactions against external details, or is this a service to verify a transaction you got from somewhere?

Reply:
And this one to `UtxoSignedTransactionChecker`.
(With its counterpart`ConsensualSignedTransactionChecker`)